packages/@react-aria/utils/src/useId.ts

```diff
@@ -87,18 +88,20 @@ export function mergeIds(idA: string, idB: string): string {
 /**
  * Used to generate an id, and after render, check if that id is rendered so we know
  * if we can use it in places such as labelledby.
+ * @param depArray - When to recalculate if the id is in the DOM.
  */
-export function useSlotId(): string {
+export function useSlotId(depArray: any[] = []): string {
   let id = useId();
-  let [resolvedId, setResolvedId] = useState(id);
-  useLayoutEffect(() => {
-    let setCurr = idsUpdaterMap.get(id);
-    if (setCurr && !document.getElementById(id)) {
-      setResolvedId(null);
-    } else {
-      setResolvedId(id);
-    }
-  }, [id]);
+  let [resolvedId, setResolvedId] = useValueEffect(id);
+  let updateId = useCallback(() => {
+    setResolvedId(function *() {
+      yield id;
+
+      yield document.getElementById(id) ? id : null;
+    });
+  }, [id, setResolvedId]);
+
+  useLayoutEffect(updateId, [id, ...depArray]);
```

Comment:
For clarification, the `...depArray` forces the `useLayoutEffect` method to get called whenever there is a change in `errorMessage`, `description` or `validationState`, correct?

Reply:
Ah, how this works is explained in `useValueEffect`: https://github.com/adobe/react-spectrum/blob/1e2b7f280a04ead83e21b27fa580eb189e5f4186/packages/%40react-spectrum/utils/src/useValueEffect.ts#L18-L21